iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/procedure/impl/pipe/task/AlterPipeProcedureV2.java

```diff
@@ -0,0 +1,299 @@
+      response.setMessage(e.getMessage());
+    }
+    if (response.getCode() != TSStatusCode.SUCCESS_STATUS.getStatusCode()) {
+      throw new PipeException(response.getMessage());
+    }
+  }
+
+  @Override
+  protected void executeFromOperateOnDataNodes(ConfigNodeProcedureEnv env) throws IOException {
+    final String pipeName = alterPipeRequest.getPipeName();
+    LOGGER.info("AlterPipeProcedureV2: executeFromOperateOnDataNodes({})", pipeName);
+
+    String exceptionMessage =
+        parsePushPipeMetaExceptionForPipe(pipeName, pushSinglePipeMetaToDataNodes(pipeName, env));
+    if (!exceptionMessage.isEmpty()) {
+      LOGGER.warn(
+          "Failed to alter pipe {}, details: {}, metadata will be synchronized later.",
+          alterPipeRequest.getPipeName(),
+          exceptionMessage);
+    }
+  }
+
+  @Override
+  protected void rollbackFromValidateTask(ConfigNodeProcedureEnv env) {
+    LOGGER.info(
+        "AlterPipeProcedureV2: rollbackFromValidateTask({})", alterPipeRequest.getPipeName());
+    // Do nothing
+  }
+
+  @Override
+  protected void rollbackFromCalculateInfoForTask(ConfigNodeProcedureEnv env) {
+    LOGGER.info(
+        "AlterPipeProcedureV2: rollbackFromCalculateInfoForTask({})",
+        alterPipeRequest.getPipeName());
+    // Do nothing
+  }
+
+  @Override
+  protected void rollbackFromWriteConfigNodeConsensus(ConfigNodeProcedureEnv env) {
+    LOGGER.info(
+        "AlterPipeProcedureV2: rollbackFromWriteConfigNodeConsensus({})",
+        alterPipeRequest.getPipeName());
+    TSStatus response;
+    try {
+      response =
+          env.getConfigManager()
+              .getConsensusManager()
+              .write(new DropPipePlanV2(alterPipeRequest.getPipeName()));
+    } catch (ConsensusException e) {
+      LOGGER.warn("Failed in the write API executing the consensus layer due to: ", e);
+      response = new TSStatus(TSStatusCode.EXECUTE_STATEMENT_ERROR.getStatusCode());
+      response.setMessage(e.getMessage());
+    }
+    if (response.getCode() != TSStatusCode.SUCCESS_STATUS.getStatusCode()) {
+      throw new PipeException(response.getMessage());
+    }
+  }
+
+  @Override
+  protected void rollbackFromOperateOnDataNodes(ConfigNodeProcedureEnv env) throws IOException {
+    LOGGER.info(
+        "AlterPipeProcedureV2: rollbackFromOperateOnDataNodes({})", alterPipeRequest.getPipeName());
+
+    // Push all pipe metas to datanode, may be time-consuming
+    String exceptionMessage =
+        parsePushPipeMetaExceptionForPipe(
+            alterPipeRequest.getPipeName(), pushPipeMetaToDataNodes(env));
+    if (!exceptionMessage.isEmpty()) {
+      LOGGER.warn(
+          "Failed to rollback alter pipe {}, details: {}, metadata will be synchronized later.",
+          alterPipeRequest.getPipeName(),
+          exceptionMessage);
+    }
+  }
+
+  @Override
+  public void serialize(DataOutputStream stream) throws IOException {
```

Comment:
In theory, we need to serialize all members, including alterPipeRequest, pipeStaticMeta, and pipeRuntimeMeta